build-output-api/isr-revalidation/.vercel/output/functions/revalidate.func/index.js

```diff
@@ -0,0 +1,27 @@
+const fetch = require('node-fetch')
+
+// The bypass token can be a randomly generated string of at least 32 characters.
+// This is meant to be *private* - DO NOT expose this value on the client-side.
+const bypassToken = '87734ad8259d67c3c11747d3e4e112d0'
+
+const deployedUrl = 'https://isr-revalidation-endangeredmassa.vercel.app'
+
+module.exports = (req, res) => {
+  fetch(deployedUrl, {
+    headers: {
+      'x-prerender-revalidate': bypassToken
+    }
+  }).then(() => {
+    res.setHeader('Content-Type', 'text/html; charset=utf-8')
+    res.end(`
+<h1>Cache Revalidated!</h1>
+<p>Redirecting you back.</p>
+<meta http-equiv="refresh" content="2; url=${deployedUrl}">
```

Comment:
If I respond with a redirect (`location` header), the page load still has the cache. I have to reload the page again to see the new value.

```
res.statusCode = 307
res.setHeader('location', '/')
res.end()
```

If I wait a couple seconds, then redirect back with a `meta http-equiv="refresh"`, the page load shows the new value.

Is this expected or am I doing something wrong here?

Reply:
The on-demand revalidate is blocking although it can take a little to purge all proxy workers/regions